lib/resilient/faraday_middleware.rb

```diff
@@ -6,13 +6,17 @@ class FaradayMiddleware < Faraday::Middleware
     VERSION = "0.1.0"
 
     def call(request_env)
-      # do something with the request
-      # request_env[:request_headers].merge!(...)
+      request_host    = request_env.url.host
+      circuit_breaker = Resilient::CircuitBreaker.get(request_host)
+
+      raise NotImplementedError unless circuit_breaker.allow_request?
```

Comment:
I'm not sure if it's preferable to raise an explicit error (TBD: that's why this is still `NotImplementedError`) or instantiate a new `Faraday::Response` error with a `503` status code.